packages/esm-patient-immunizations-app/src/immunizations/immunization-history-card.component.tsx

```diff
@@ -31,25 +35,40 @@ const ImmunizationHistoryCard: React.FC<ImmunizationHistoryCardProps> = ({ error
     { key: 'doses', header: t('doses', 'Doses') },
   ];
 
+  const getNextDoseStatus = (nextDoseDate: string): 'DUE' | 'NOT_DUE' => {
+    const today = dayjs().startOf('day');
+    const nextDate = dayjs(nextDoseDate).startOf('day');
+
+    return today.isSameOrAfter(nextDate) ? 'DUE' : 'NOT_DUE';
+  };
```

Comment:
I would hoist this function outside of the React component since it doesn't depend at all on anything in the component.